notebook_utils/berdl_notebook_utils/spark/connect_server.py

```diff
@@ -0,0 +1,278 @@
+"""
+Spark Connect server management for BERDL notebook environments.
+
+This module provides utilities for starting and managing user-specific Spark Connect servers
+that run alongside notebook kernels, enabling remote Spark session connectivity.
+"""
+
+import logging
+import os
+import shutil
+import subprocess
+import time
+from pathlib import Path
+from typing import Optional
+
+from ..berdl_settings import BERDLSettings, get_settings
+
+logger = logging.getLogger(__name__)
+
+
+class SparkConnectServerConfig:
+    """Configuration for Spark Connect server."""
+
+    def __init__(self, settings: Optional[BERDLSettings] = None):
+        """
+        Initialize Spark Connect server configuration.
+
+        Args:
+            settings: BERDLSettings instance. If None, loads from environment.
+        """
+        if settings is None:
+            settings = get_settings()
+        self.settings = settings
+        self.username = settings.USER
+
+        # Spark directories
+        self.spark_home = os.environ.get("SPARK_HOME", "/usr/local/spark")
+        self.user_spark_dir = Path.home() / ".spark"
+        self.user_conf_dir = self.user_spark_dir / "conf"
+        self.spark_event_log_dir = self.user_spark_dir / "spark-events"
+        self.connect_server_log_dir = self.user_spark_dir / "connect-server-logs"
+
+        # Configuration files
+        self.spark_defaults_path = self.user_conf_dir / "spark-defaults.conf"
+        self.template_path = os.environ.get("SPARK_CONNECT_DEFAULTS_TEMPLATE", "/configs/spark-defaults.conf.template")
+
+        # Server configuration
+        self.spark_connect_port = int(os.environ.get("SPARK_CONNECT_PORT", "15002"))
+        self.spark_master_url = os.environ.get("SPARK_MASTER_URL", "spark://spark-master:7077")
```

Comment:
Shouldn't these env vars be coming from the settings?

Reply:
👍 sure.